src/Store.js

```diff
@@ -921,17 +921,25 @@ class Store {
             return Promise.reject(new Error(this.genericErrorMessage))
           }
 
+          if (!json.errors) {
+            return Promise.reject(new Error(this.genericErrorMessage))
+          }
+
           // Add all errors from the API response to the record(s).
           // This is done by comparing the pointer in the error to
           // the request.
-          json.errors.forEach((error) => {
-            const { index, key } = parseErrorPointer(error)
-            if (key != null) {
-              const errors = recordsArray[index].errors[key] || []
-              errors.push(error)
-              recordsArray[index].errors[key] = errors
-            }
-          })
+          if (Array.isArray(json.errors)) {
+            json.errors.forEach((error) => {
+              const { index, key } = parseErrorPointer(error)
+              if (key != null) {
+                let errors = recordsArray[index].errors[key] || []
+                errors.push(error)
+                recordsArray[index].errors[key] = errors
+              }
+            })
+          } else {
+            return Promise.reject(new TypeError('Top level errors in response are not an array.'))
+          }
```

Comment:
[non-blocking] I'm just realizing we should probably early-return based on it not being an array, that way the main code loop isn't inside the `if`.